### PR TITLE
[SHIP-3903]Remove WS requirement for data feeds soak tests

### PR DESCRIPTION
**Pull request by davidcauchi**
[SHIP-3903](https://smartcontract-it.atlassian.net/browse/SHIP-3903)

Switch from subscription to polling for logs to remove the requirement of having a WS endpoint.

[SHIP-3903]: https://smartcontract-it.atlassian.net/browse/SHIP-3903?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kalverra — DISMISSED** [review]

**Review by joaoluisam — DISMISSED** [review]
LGTM

File: integration-tests/testsetups/ocr.go [hunk 1]
```diff
@@ -824,70 +823,106 @@ func (o *OCRSoakTest) setFilterQuery() {
...
-						backoff = time.Duration(math.Min(float64(backoff)*2, float64(30*time.Second)))
+							Uint64("Round ID", answerUpdated.RoundId.Uint64()).
+							Int64("Answer", answerUpdated.Current.Int64()).
+							Msg("Answer Updated Event")
 					}
 				}
+
+				// Track last checked block number and update starting block number
+				lastCheckedBlockNum = latestBlock
+				startingBlockNum = latestBlock + 1
```
Comment (simsonraj) [inline integration-tests/testsetups/ocr.go]: Minor nit: can we just repurpose startingBlockNum to check for the lastCheckedBlockNum. Not a big deal but good for readability

File: integration-tests/testsetups/ocr.go [hunk 2]
```diff
@@ -824,70 +823,106 @@ func (o *OCRSoakTest) setFilterQuery() {
...
 
 	go func() {
+		// TODO: Make this configurable
+		pollInterval := time.Second * 30
+		ticker := time.NewTicker(pollInterval)
+		defer ticker.Stop()
+		o.log.Info().Msg("Start Polling for Answer Updated Events")
 		for {
 			select {
-			case event := <-eventLogs:
```
Comment (simsonraj) [inline integration-tests/testsetups/ocr.go:841]: Do you think its worth keeping the WSS option also?

**Review by simsonraj — DISMISSED** [review]

File: integration-tests/testsetups/ocr.go (hunk 2 above)
Comment (davidcauchi) [reply integration-tests/testsetups/ocr.go:841]: I would move onto polling. Seems more reliable and not prone to disconnections.

File: integration-tests/testsetups/ocr.go [hunk 3]
```diff
@@ -824,70 +823,105 @@ func (o *OCRSoakTest) setFilterQuery() {
...
-	if err != nil {
-		return err
-	}
+// pollingOCREvents Polls the blocks for OCR events and logs them to the test logger
+func (o *OCRSoakTest) pollingOCREvents(endTest <-chan time.Time) error {
+
+	// Keep track of the last processed block number
+	processedBlockNum := o.startingBlockNum - 1
 
 	go func() {
```
Comment (jmank88) [inline integration-tests/testsetups/ocr.go]: Out of scope, but this goroutine appears to be untracked.

File: integration-tests/testsetups/ocr.go [hunk 4]
```diff
@@ -824,70 +823,105 @@ func (o *OCRSoakTest) setFilterQuery() {
...
+		// TODO: Make this configurable
```
Comment (jmank88) [inline integration-tests/testsetups/ocr.go]: For this PR or follow-up? Is there a ticket to link?

File: integration-tests/testsetups/ocr.go (hunk 1 above)
Comment (davidcauchi) [reply integration-tests/testsetups/ocr.go]: Thanks for the suggestion. Wanted to have a clear separation between the initial run and subsequent runs but it does not add that much value. Simpler code is always better.

File: integration-tests/testsetups/ocr.go (hunk 4 above)
Comment (davidcauchi) [reply integration-tests/testsetups/ocr.go]: Intention was for a follow up. Just created SHIP-3973 to cover it.

File: integration-tests/testsetups/ocr.go (hunk 3 above)
Comment (davidcauchi) [reply integration-tests/testsetups/ocr.go]: I have updated how this is handled and it is now tracked. Thanks for pointing it out.

**Review by kalverra — APPROVED** [review]

**Review by joaoluisam — APPROVED** [review]

**Pull request merged**